source/mongodb/tables/role-configuration.rst

```diff
@@ -36,82 +41,159 @@
+       undefined and use ``additional_fields``.
 
    * - | ``write``
        | Expression
-       | *Default:* ``false``
-     - An :ref:`expression <expressions>` that evaluates to ``true`` if the
+       | *Default:* ``undefined``
+     - An :ref:`expression <expressions>` that evaluates to true if the
        role has permission to add, modify, or remove all fields in the document.
 
+       If this role is to be :ref:`sync compatible <sync-compatible-roles>`,
+       the expression must be a boolean literal (either ``true`` or ``false``).
+
        Document-level write permissions take priority over any field-level
        permissions. If a role has document-level ``write`` permissions, it
-       applies to all fields in the document and cannot be overridden.
+       applies to all fields in the document. Write permissions specified by
+       ``fields`` or ``additional_fields`` do not override document-level
+       ``write`` permissions.
        
-       To define a default fallback alongside field-level rules, use
-       ``additional_fields`` instead.
+       To define a default fallback alongside field-level rules, leave ``write``
+       undefined and use ``additional_fields``.
        
+       Note that you can use expansions like :json-expansion:`%%root` and
+       :json-expansion:`%%prevRoot` in ``write`` JSON expressions.
+
        .. important:: Implicit Read Permission
 
           Any time a role has ``write`` permission for a particular
           scope it also has ``read`` permission even if that is not
           explicitly defined.
-       
-       .. note:: MongoDB Expansions
-          
-          You can use MongoDB expansions, like :json-expansion:`%%root`
-          and :json-expansion:`%%prevRoot`, in ``write`` JSON
-          expressions.
 
    * - | ``insert``
        | Expression
        | *Default:* ``true``
      - An :ref:`expression <expressions>` that evaluates to
        ``true`` if the role has permission to insert a new document into the
        collection.
-
-       .. note::
-          
-          Document-level ``insert`` permission does *not* imply that a
-          role can insert any document. The role must also have
-          ``write`` permission for all fields in an inserted document
-          for the insert to succeed.
+       
+       App Services only evaluates this expression for insert operations and
+       only after determining that the role has ``write`` permission for all
+       fields in the new document.
 
    * - | ``delete``
        | Expression
        | *Default:* ``true``
-     - An :ref:`expression <expressions>` that evaluates to ``true`` if the
+     - An :ref:`expression <expressions>` that evaluates to true if the
        role has permission to delete a document from the collection.
 
+       App Services only evaluates this expression for delete operations and
+       only after determining that the role has ``write`` permission for all
```

Comment:
We actually don't perform any checks involving `write` / field-level `write` perms on deleting a document (see https://docs.google.com/document/d/1tK_vKysOVEb5VdSPBxu-0NXB_WZZ5TnjA0fNx7HpKZQ/edit#heading=h.xnyhe9mxtwef for more info)

Note we do however perform the document_filters checks before checking inserts/deletes

Reply:
Correction: just double checked and we actually do the document_filters checks *after* checking inserts/deletes.

Reply:
We should probably also update the diagrams as well to capture that delete permission checks are a bit different (they don't check to see if the fields are writable before actually deleting the document)